doc/prelimsMac.rst

```diff
@@ -0,0 +1,204 @@
+Preliminary Steps for macOS
+===========================
+
+1) Install Python 3.6+
+----------------------
+Check your current python version(s) by entering the following commands into the command line: 
+
+.. code-block:: shell
+
+   python --version
+   python3 --version
+   python3.6 --version
+
+Running **macOS** install/update Python 3.0+ (here using the reccomended `Homebrew <https://brew.sh/>`__ package manager):
+
+.. code-block:: shell
+
+   brew install python3
+   brew upgrade python3 
+
+A .bashrc file is a shell script that Bash runs whenever it is started or executed by the user. The installation instructions for other Linux distributions involve adding lines to the built-in .bashrc, and creating one is neccesary on macOS for aliases and path adjustments. However, you must first create the .bashrc file as it is not included by default on Mac systems. To create and open a plain text .bashrc file, enter the following into your command line:
+
+.. code-block:: shell
+   
+   touch ~/.bashrc
+   open ~/.bashrc
+
+This file will not appear in your home directory unless you enter 'ls -a' in the terminal to show hidden files. Although not strictly neccesary, paste the two lines below into your new .bashrc file; these lines map the python and pip commands to their Python 3.6+ counterparts (i.e. 
+*python helloworld.py* will be executed using Python 3.6+).
+
+.. code-block:: shell
+      
+   alias python=python3
+   alias pip=pip3
+
+To set these changes into effect, enter the following command in the command line. You may also wish to have the command below run automatically on the startup of your shell. To do this paste the command into the the *Startup - Run command* field in Terminal>Preferences>Profiles>Shell.
+
+.. code-block:: shell
+      
+   source ~/.bashrc
+
+
+2) Install gcc and virtualenvwrapper
+------------------------------------
+
+a) Install gcc(maybe now new clang)
+***********************************
+
+Gcc (`GNU Compiler Collection <https://gcc.gnu.org/>`__) is a compiler system that supports multiple languages. Although an older version is most already installed on your Mac, install the newest version using brew:
```

Comment:
macOS's default "GCC" is actually clang. And like I said before, I would strongly advise against using GCC on the Mac since it gave me many problems. Switching to a more recent version of clang fixed many of my issues.